codec/raw/codec.go

```diff
@@ -0,0 +1,63 @@
+// Package raw implements IPLD's raw codec, which simply writes and reads a Node
+// which can be represented as bytes.
+//
+// The codec can be used with any node which supports AsBytes and AssignBytes.
+// In general, it only makes sense to use this codec on a plain "bytes" node
+// such as github.com/ipld/go-ipld-prime/node/basic.Prototype.Bytes.
+package raw
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	ipld "github.com/ipld/go-ipld-prime"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+)
+
+var (
+	_ cidlink.MulticodecDecoder = Decoder
+	_ cidlink.MulticodecEncoder = Encoder
+)
+
+// TODO(mvdan): make go-ipld-prime use go-multicodec soon
+const rawMulticodec = 0x55
+
+func init() {
+	cidlink.RegisterMulticodecDecoder(rawMulticodec, Decoder)
+	cidlink.RegisterMulticodecEncoder(rawMulticodec, Encoder)
+}
+
+type byteAccesor interface {
+	Bytes() []byte
+}
+
+func Decoder(am ipld.NodeAssembler, r io.Reader) error {
+	// We need r's bytes, and r is often just a buffer.
+	// So if we can get them directly without ReadAll's copying,
+	// just do that and also save an allocation.
+	// Nodes are immutable, so we won't modify the bytes.
+	// We assume that the incoming buffer won't get modified either.
```

Comment:
I think we should move this to the actual docs block on the function.  (I'm probably guilty of putting too many tidbits like this in comments too, but I should also quit it.)

I'm trying to get in the habit of using phrasing about "it is not lawful to xyz" ([example](https://github.com/ipld/go-ipld-prime/blob/7d918468f40d06d8a64a2fb6c162134da497bebb/path.go#L152)) for things like this mutability contract that we can't really enforce with the compiler.  Just to have _some_ convention for talking about this semi-formally.

Reply:
Fair enough. I thought about it initially, and thought that some of these details are too internal for the godoc. But I think you're right, it's still better than no godoc.